server/sonar-process/src/main/java/org/sonar/process/StopWatcher.java

```diff
@@ -45,23 +45,19 @@ public StopWatcher(ProcessCommands commands, Stoppable stoppable) {
 
   @Override
   public void run() {
-    commands.prepare();
-    try {
-      while (watching) {
-        if (commands.askedForStop()) {
-          LoggerFactory.getLogger(getClass()).info("Stopping process");
-          stoppable.stopAsync();
+    while (watching) {
+      commands.ping();
```

Comment:
If my understanding is correct, it's the thread that monitor if there is an askedForStop every 500ms. I though it's the best way to both monitor the flag and update the field containing the last ping command. I think that we do not need another thread just for pinging.